bugsnag-android-core/src/main/java/com/bugsnag/android/Journal.java

```diff
@@ -0,0 +1,170 @@
+package com.bugsnag.android;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.dslplatform.json.DslJson;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A journal represents a series of commands that will manipulate a document.
+ * Sometimes it's more efficient to store a base document and a list of modifications
+ * to me made to it rather than re-serialize the entire document after every modification.
+ */
+public class Journal {
```

Comment:
Should these classes be exposed as part of our public API? This feels like something to avoid unless it's absolutely necessary. If this is unavoidable, then this could be moved to the `com.bugsnag.android.internal` package which be convention is not part of the API.

Reply:
I've now moved them to internal, and changed the implementation to Kotlin.